did_peer/src/key/key.rs

```diff
@@ -0,0 +1,244 @@
+use crate::error::DidPeerError;
+
+use super::SupportedKeyType;
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Key {
+    key_type: SupportedKeyType,
+    key: Vec<u8>,
+}
+
+impl Key {
+    pub fn new(key: Vec<u8>, key_type: SupportedKeyType) -> Result<Self, DidPeerError> {
+        // If the key is a multibase key coming from a verification method, for some reason it is also
+        // multicodec encoded, so we need to strip that. But should it be?
+        let key = Self::strip_multicodec_prefix_if_present(key, &key_type);
+        Ok(Self { key_type, key })
+    }
+
+    pub fn key_type(&self) -> &SupportedKeyType {
+        &self.key_type
+    }
+
+    pub fn key(&self) -> &[u8] {
+        self.key.as_ref()
+    }
+
+    pub fn prefixed_key(&self) -> Vec<u8> {
+        let code = self.key_type().into();
+        let mut buffer = [0u8; 10];
+        let bytes = unsigned_varint::encode::u64(code, &mut buffer);
+        let mut prefixed_key = bytes.to_vec();
+        prefixed_key.extend_from_slice(&self.key);
+        prefixed_key
+    }
+
+    pub fn fingerprint(&self) -> String {
+        multibase::encode(multibase::Base::Base58Btc, &self.prefixed_key())
+    }
+
+    pub fn prefixless_fingerprint(&self) -> String {
```

Comment:
OK, see no problem with that

Reply:
Actually, this implementation relies on the base of `fingerprint`, which is internal to `Key` (i.e. may change). Also, there are multiple places where it can be used.

Reply:
roger that!